text/0000-build-std/4-stage-1a.md

```diff
@@ -0,0 +1,1003 @@
+upgrading the toolchain version.
+
+See
+[*Why use the lockfile of the `rust-src` component?*][rationale-lockfile]
+
+↩ [*Vendored `rust-src`*][vendored-rust-src]
+
+### Why not check if `rust-src` has been modified?
+[rationale-src-modifications]: #why-not-check-if-rust-src-has-been-modified
+
+It is likely that any protections implemented to check that the sources in
+`rust-src` have not been modified could be trivially bypassed.
+
+Any crate that depends on `rust-src` having been modified would not be usable
+when published to crates.io as the required modifications will obviously not be
+included.
+
+↩ [*Vendored `rust-src`*][vendored-rust-src]
+
+### Why allow building from the sysroot with implied `RUSTC_BOOTSTRAP`?
+[rationale-implied-bootstrap]: #why-allow-building-from-the-sysroot-with-implied-rustc_bootstrap
+
+Cargo needs to be able to build the standard library crates, which inherently
+require a nightly toolchain. It could set `RUSTC_BOOTSTRAP` internally to do
+this with a stable toolchain, however this is a shared requirement with other
+build systems that wish to build an unmodified standard library and want to work
+on stable toolchains.
+
+For example, Rust's project goal to enable Rust for Linux to build using only a
+stable toolchain would require that it be possible to build `core` without
+nightly.
+
+It is not sufficient for rustc to special-case the `core`, `alloc` and `std`
+crate names as, when being built as part of the standard library, dependencies
+of the standard library also use unstable features and it is not practical to
+special-case all of these crates.
+
+↩ [*Building the standard library on a stable toolchain*][building-the-standard-library-on-a-stable-toolchain]
+
+### Why not use weak linkage for `compiler-builtins/mem` symbols?
+[rationale-no-weak-linkage]: #why-not-use-weak-linkage-for-compiler-builtinsmem-symbols
+
+Since [compiler-builtins#411], the relevant symbols in `compiler_builtins`
+already have weak linkage. However, it is nevertheless not possible to simply
+remove the `mem` feature and have the symbols always be present.
+
+Some targets, such as those based on MinGW, do not have sufficient support for
+weak definitions (at least with the default linker). Furthermore, weak linkage
+has precedence over shared libraries and the symbols of a dynamically-linked
+`libc` should be preferred over `compiler_builtins`'s symbols.
+
+↩ [*`compiler-builtins-mem`*][compiler-builtins-mem]
+
+### Why not globally cache builds of the standard library?
+[rationale-caching]: #why-not-globally-cache-builds-of-the-standard-library
+
+The standard library is no different than regular dependencies in being able to
+benefit from global caching of dependency builds. A generic proposal for global
+dependency caching could support the standard library. It is out-of-scope of
+this proposal to propose a special-cased mechanism for this that applies only to
+the standard library.
+
```

Comment:
Should we link out to https://github.com/rust-lang/cargo/issues/5931 for solving this?

Reply:
Makes sense, I had looked for an issue for that but couldn't find it.

Reply:
Done in https://github.com/davidtwco/rfcs/pull/2/commits/5aaad19e565fedff1423a9fc17fb854a2d3ead39